draft-ietf-masque-connect-udp.md

```diff
@@ -276,34 +276,59 @@ the client (e.g., "connect-udp-version: 2"). Sending this header is RECOMMENDED
 but not required.
 
 
-# Encoding of Proxied UDP Packets {#datagram-encoding}
-
-UDP packets are encoded using HTTP Datagrams {{HTTP-DGRAM}} with the
-UDP_PAYLOAD HTTP Datagram Format Type (see value in {{iana-format-type}}). When
-using the UDP_PAYLOAD HTTP Datagram Format Type, the payload of a UDP packet
-(referred to as "data octets" in {{UDP}}) is sent unmodified in the "HTTP
-Datagram Payload" field of an HTTP Datagram.
-
-In order to use HTTP Datagrams, the client will first decide whether or not it
-will attempt to use HTTP Datagram Contexts and then register its context ID (or
-lack thereof) using the corresponding registration capsule, see {{HTTP-DGRAM}}.
-
-When sending a registration capsule using the "Datagram Format Type" set to
-UDP_PAYLOAD, the "Datagram Format Additional Data" field SHALL be empty.
-Servers MUST NOT register contexts using the UDP_PAYLOAD HTTP Datagram Format
-Type. Clients MUST NOT register more than one context using the UDP_PAYLOAD
-HTTP Datagram Format Type. Endpoints MUST NOT close contexts using the
-UDP_PAYLOAD HTTP Datagram Format Type. If an endpoint detects a violation of
-any of these requirements, it MUST abort the stream.
+# Context Identifiers {#context-id}
+
+This protocol allows future extensions to exchange HTTP Datagrams which carry
+different semantics from UDP payloads. Some of these extensions can augment UDP
+payloads with additional data, while others can exchange data that is completely
+separate from UDP payloads. In order to accomplish this, all HTTP Datagrams
+associated with UDP Proxying request streams start with a context ID, see
+{{format}}.
+
+Context IDs are 62-bit integers (0 to 2<sup>62</sup>-1). The context ID value of
+zero is reserved for UDP payloads, while non-zero values are dynamically
+allocated: non-zero even-numbered context IDs are client-initiated, and
```

Comment:
I prefer the term initiated since that maps to what QUIC uses to refer to streams. If that's OK with you I'm going to declare this editorial and that we can discuss it with the WG instead of in the design team.

Reply:
That being the case, please be consistent and use "initiate" rather than "allocate" later.  I was just trying to make the terms you use in the definition consistent with the usage later.  (In the direction I prefer, but you do get editor's privilege there.)

Reply:
Fair enough, you've convinced me. Switched to client-allocated.